Formula/goredo.rb

```diff
@@ -0,0 +1,27 @@
+class Goredo < Formula
+  desc "Go implementation of djb's redo, a Makefile replacement that sucks less"
+  homepage "http://www.goredo.cypherpunks.ru/"
+  url "http://www.goredo.cypherpunks.ru/download/goredo-0.11.0.tar.zst"
+  version "0.11.0"
+  sha256 "4608b75bc8a3dfb62ef305b5eed0f4b28ad39f64ae46baceea8100d90a8666f9"
+  license "GPL-3.0-only"
+
+  depends_on "go" => :build
+  depends_on "zstd" => :build
+
+  def install
+    goredo_prefix = "goredo-#{version}"
+    system "tar", "--use-compress-program", "unzstd", "-xvf", "#{goredo_prefix}.tar.zst"
+    cd goredo_prefix do
+      ENV["GOPATH"] = Pathname.pwd
+      system "go", "build", *std_go_args, "go.cypherpunks.ru/goredo"
+    end
+    cd bin do
+      system "./goredo", "-symlinks"
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/redo-ifchange --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I added a test:
* redo command existence
* -no-progress -- tests that it is goredo implementation, having that option
* ability of .do file searching
* ability of .do file running with the working shell (if OS does not have /bin/sh, then it will fail, as an example)